test/framework/control_plane.go

```diff
@@ -34,213 +34,276 @@ import (
+	ControlPlane *controlplanev1.KubeadmControlPlane
+}
 
-		By("waiting for the workload nodes to exist")
-		Eventually(func() ([]v1.Node, error) {
-			workloadClient, err := input.Management.GetWorkloadClient(ctx, input.Cluster.Namespace, input.Cluster.Name)
-			if err != nil {
-				return nil, errors.Wrap(err, "failed to get workload client")
-			}
-			nodeList := v1.NodeList{}
-			if err := workloadClient.List(ctx, &nodeList); err != nil {
-				return nil, err
-			}
-			return nodeList.Items, nil
-		}, input.CreateTimeout, 10*time.Second).Should(HaveLen(int(*machineDeployment.Spec.Replicas)))
+// WaitForKubeadmControlPlaneMachinesToExist will wait until all control plane machines exist.
+func WaitForKubeadmControlPlaneMachinesToExist(ctx context.Context, input WaitForKubeadmControlPlaneMachinesToExistInput, intervals ...interface{}) {
+	By("waiting for all control plane nodes to exist")
+	inClustersNamespaceListOption := client.InNamespace(input.Cluster.Namespace)
+	// ControlPlane labels
+	matchClusterListOption := client.MatchingLabels{
+		clusterv1.MachineControlPlaneLabelName: "",
+		clusterv1.ClusterLabelName:             input.Cluster.Name,
 	}
 
-	By("waiting for all machines to be running")
-	inClustersNamespaceListOption := client.InNamespace(input.Cluster.Namespace)
-	matchClusterListOption := client.MatchingLabels{clusterv1.ClusterLabelName: input.Cluster.Name}
-	Eventually(func() (bool, error) {
-		// Get a list of all the Machine resources that belong to the Cluster.
+	Eventually(func() (int, error) {
 		machineList := &clusterv1.MachineList{}
-		if err := mgmtClient.List(ctx, machineList, inClustersNamespaceListOption, matchClusterListOption); err != nil {
-			return false, err
-		}
-		for _, machine := range machineList.Items {
-			if machine.Status.Phase != string(clusterv1.MachinePhaseRunning) {
-				return false, errors.Errorf("machine %s is not running, it's %s", machine.Name, machine.Status.Phase)
-			}
+		if err := input.Lister.List(ctx, machineList, inClustersNamespaceListOption, matchClusterListOption); err != nil {
+			fmt.Println(err)
+			return 0, err
 		}
-		return true, nil
-	}, input.CreateTimeout, eventuallyInterval).Should(BeTrue())
-	// wait for the control plane to be ready
+		return len(machineList.Items), nil
+	}, intervals...).Should(Equal(int(*input.ControlPlane.Spec.Replicas)))
+}
+
+// WaitForControlPlaneToBeReadyInput is the input for WaitForControlPlaneToBeReady.
+type WaitForControlPlaneToBeReadyInput struct {
+	Getter       Getter
+	ControlPlane *controlplanev1.KubeadmControlPlane
+}
+
+// WaitForControlPlaneToBeReady will wait for a control plane to be ready.
+// TODO(chuckha): Once we implement control plane Ready, then we should update this to wait actually wait for ready.
+// TODO(chuckha): In the meantime this uses initialized as a placeholder for Ready.
+func WaitForControlPlaneToBeReady(ctx context.Context, input WaitForControlPlaneToBeReadyInput, intervals ...interface{}) {
```

Comment:
This can be confused with waiting for a Kubernetes control-plane to be ready, but I don't have a better suggestion ATM

Reply:
Ah, yeah, I might rename to ClusterAPIManagedControlPlaneReady, but we can tackle that later since we haven't even implemented control plane ready yet

Reply:
+1 to addressing later, let's unblock the additional control plane work and then revisit later.